test/test_srv_polling.py

```diff
@@ -230,6 +230,50 @@ def erroring_seedlist():
             raise ConfigurationError
         self._test_recover_from_initial(erroring_seedlist)
 
+    def test_10_all_dns_selected(self):
+        response = [("localhost.test.build.10gen.cc", 27017),
+                    ("localhost.test.build.10gen.cc", 27017),
+                    ("localhost.test.build.10gen.cc", 27019),
+                    ("localhost.test.build.10gen.cc", 27020)]
+
+        def nodelist_callback():
+            return response
+        with SrvPollingKnobs(ttl_time=WAIT_TIME,
+                             min_srv_rescan_interval=WAIT_TIME):
+            client = MongoClient(self.CONNECTION_STRING, srvMaxHosts=0)
+            with SrvPollingKnobs(nodelist_callback=nodelist_callback):
+                self.assert_nodelist_change(response, client)
+                sleep(2*common.MIN_SRV_RESCAN_INTERVAL)
+                expected_topology = {
+                    ("localhost.test.build.10gen.cc", 27017),
+                    ("localhost.test.build.10gen.cc",  27019),
+                    ("localhost.test.build.10gen.cc",  27020)}
+                final_topology = set(
+                    client.topology_description.server_descriptions())
+                self.assertSetEqual(final_topology, final_topology |
```

Comment:
please remove :
```python
                expected_topology = {
                    ("localhost.test.build.10gen.cc", 27017),
                    ("localhost.test.build.10gen.cc",  27019),
                    ("localhost.test.build.10gen.cc",  27020)}
                final_topology = set(
                    client.topology_description.server_descriptions())
                self.assertSetEqual(final_topology, final_topology |
```

These lines are all redundant because `self.assert_nodelist_change(response, client)` already does the same assertion.

Reply:
I think that removing self.assert_nodelist_change would be better because the code I wrote actually checks for if it's a subset, which is what the spec says to do.

Reply:
Done.